src/script/components/Conversation/ConversationCells/CellsPagination/CellsPagination.tsx

```diff
@@ -0,0 +1,175 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import React, {useCallback} from 'react';
+
+import {FlexBox, Bold, Link, IconButton, Select} from '@wireapp/react-ui-kit';
+
+import * as Icon from 'Components/Icon';
+
+import {
+  pagesContainerStyles,
+  numberStyles,
+  numberActiveStyles,
+  arrowButtonStyles,
+  arrowNextIconStyles,
+  arrowPreviousIconStyles,
+  containerStyles,
+  selectorStyles,
+  pageSelectorContainer,
+} from './CellsPagination.styles';
+
+import {useCellsStore} from '../common/useCellsStore/useCellsStore';
+
+interface CellsPaginationProps {
+  currentPage?: number;
+  numberOfPages?: number;
+  firstRow?: number;
+  lastRow?: number;
+  totalRows?: number;
+  goPage: (pageIndex: number) => void;
+  [key: string]: any;
```

Comment:
we shouldn't add a way to inject any props we want, let's declare only those we're gonna use